pkg/k8shandler/defaults.go

```diff
@@ -10,6 +10,23 @@ const (
 	modeUnique    = "unique"
 	modeSharedOps = "shared_ops"
 	defaultMode   = modeSharedOps
+
+	defaultMasterCPULimit     = "100m"
+	defaultMasterCPURequest   = "100m"
+	defaultCPULimit           = "4000m"
+	defaultCPURequest         = "100m"
+	defaultMemoryLimit        = "4Gi"
+	defaultMemoryRequest      = "1Gi"
+	elasticsearchDefaultImage = "quay.io/openshift/origin-logging-elasticsearch5"
+
+	maxMasterCount = 3
+
+	elasticsearchCertsPath  = "/etc/openshift/elasticsearch/secret"
+	elasticsearchConfigPath = "/usr/share/java/elasticsearch/config"
+	heapDumpLocation        = "/elasticsearch/persistent/heapdump.hprof"
+
+	defaultDiskWatermarkLow  = float64(85.0)
+	defaultDiskWatermarkHigh = float64(90.0)
```

Comment:
Basically, we should not rely on hardcoded default values like this. These should be pulled from Elasticsearch.
If you go like this:
```
curl [...] <es_node>/_cluster/settings?include_defaults=true&filter_path=defaults.cluster.routing.allocation.disk
```
You get this (assuming ES 5.x):
```json
{
  "defaults": {
    "cluster": {
      "routing": {
        "allocation": {
          "disk": {
            "threshold_enabled": "true",
            "watermark": {
              "low": "85%",
              "high": "90%"
            },
            "include_relocations": "true",
            "reroute_interval": "60s"
          }
        }
      }
    }
  }
}
```
Also notice, the value `threshold_enabled` is important too :-), if it is set to `false` then the watermark checks/logic is turned off.

To make it more complicated users can override these values on the fly at the `transient` or/and `persistent` levels. You get the values from these levels using:
```
curl [...] <es_node>/_cluster/settings?include_defaults=true&filter_path=transient.cluster.routing.allocation.disk
curl [...] <es_node>/_cluster/settings?include_defaults=true&filter_path=permanent.cluster.routing.allocation.disk
                                                       ### Check this ---^^^^^^^^^
```
Please check [Precedence of settings](https://www.elastic.co/guide/en/elasticsearch/reference/5.6/cluster-update-settings.html#_precedence_of_settings) to see which settings value are used when it is set at several levels at the same time. Notice the values set in `elasticsearch.yml` file are applied at the `default` level.

Reply:
Oh... and to make it even more fun, the values, as you already learned, can be specified either in pcts or bytes (and of course when set as bytes then they need to be treated differently...).

BTW, can not we rather rely on Prometheus metrics here? The Elasticsearch Prometheus plugin already pulls these thresholds since v6.5.4 (so you wouldn't have to deal with the complicated `default/permanent/transient` thing and the dynamic nature of it). See https://github.com/vvanholl/elasticsearch-prometheus-exporter/pull/145

Reply:
is it `permanent` or `persistent`?
Also, that is just the default value used... i query ES for those values here: [1]

+1 to that precedence of settings... I was actually wondering that. I'll also add a check to make sure that `threshold_enabled` is set to true in case a user updated that on the fly.

[1] https://github.com/openshift/elasticsearch-operator/pull/92/commits/0d545063545667a123a8cb7c6f331ade603bf507#diff-b9c2245ba513eb7cad4b725b41f801f2R106

Reply:
> BTW, can not we rather rely on Prometheus metrics here? The Elasticsearch Prometheus plugin already pulls these thresholds since v6.5.4 (so you wouldn't have to deal with the complicated default/permanent/transient thing and the dynamic nature of it). See vvanholl/elasticsearch-prometheus-exporter#145

I defer this to @jcantrill

Reply:
I believe it depends on what provides us the least amount of work.  If we can scrape our own metrics endpoint and it unifies these alternate code paths, seems to me it would make more sense to figure out how to interpret the prometheus endpoint once instead of the various alternatives.